crates/ruff_cli/src/commands/format.rs

```diff
@@ -695,11 +696,11 @@ pub(super) fn warn_incompatible_formatter_settings(
     pyproject_config: &PyprojectConfig,
     resolver: Option<&Resolver>,
 ) {
+    // First, collect all rules that are incompatible regardless of the linter-specific settings.
+    let mut incompatible_rules = BTreeSet::new();
```

Comment:
What's the motivation for using a `BTreeSet` when we still need to sort the rules on line 722? Why not use a regular `HashSet` (or create a wrapper around `BTreeSet` with a `insert(Rule)` method that inserts the formatted name)